cmd/temporalite/main_integ_test.go

```diff
@@ -0,0 +1,138 @@
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+)
+
+func TestMTLSConfig(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Run ephemerally in the mtls example folder using the config there. This
+	// does expect port 7233 to be free.
+	_, thisFile, _, _ := runtime.Caller(0)
+	args := []string{
+		"run", filepath.Dir(thisFile),
+		"start",
+		"--ephemeral",
+		"--config", ".",
+		"--namespace", "default",
+		"--log-format", "pretty",
+	}
+	if !testing.Verbose() {
+		args = append(args, "--log-level", "warn")
+	}
+	cmd := exec.CommandContext(ctx, "go", args...)
+	mtlsDir := filepath.Join(thisFile, "../../../internal/examples/mtls")
+	cmd.Dir = mtlsDir
+	cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
+	t.Logf("Running go with args %v in %v", args, cmd.Dir)
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := kill(cmd); err != nil {
+			t.Logf("Process kill failed: %v", err)
+		}
+	}()
+
+	// Load client cert/key for auth
+	clientCert, err := tls.LoadX509KeyPair(
+		filepath.Join(mtlsDir, "client-cert.pem"),
+		filepath.Join(mtlsDir, "client-key.pem"),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Load server cert for CA check
+	serverCAPEM, err := os.ReadFile(filepath.Join(mtlsDir, "server-ca-cert.pem"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	serverCAPool := x509.NewCertPool()
+	serverCAPool.AppendCertsFromPEM(serverCAPEM)
+
+	// Build client options and try to connect client every 200ms for 10s
+	var options client.Options
+	options.ConnectionOptions.TLS = &tls.Config{
+		Certificates: []tls.Certificate{clientCert},
+		RootCAs:      serverCAPool,
+	}
+	var c client.Client
+	for i := 0; i < 50; i++ {
+		if c, err = client.Dial(options); err == nil {
+			break
+		}
+		time.Sleep(200 * time.Millisecond)
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Make a call
+	resp, err := c.WorkflowService().DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: "default",
+	})
+	if err != nil {
+		log.Fatal(err)
+	} else if resp.NamespaceInfo.State != enums.NAMESPACE_STATE_REGISTERED {
+		log.Fatalf("Bad state: %v", resp.NamespaceInfo.State)
+	}
+}
+
+func kill(cmd *exec.Cmd) error {
+	// Have to use taskkill on Windows to avoid custom ctrl+c code
+	if runtime.GOOS == "windows" {
+		k := exec.Command("TASKKILL", "/T", "/F", "/PID", strconv.Itoa(cmd.Process.Pid))
+		k.Stdout, k.Stderr = os.Stdout, os.Stderr
+		if err := k.Run(); err != nil {
+			return fmt.Errorf("taskkill failed: %w", err)
+		}
+	} else if err := cmd.Process.Signal(syscall.SIGTERM); err != nil {
+		return fmt.Errorf("kill failed: %w", err)
+	}
+	if _, err := cmd.Process.Wait(); err != nil {
+		return fmt.Errorf("wait failed: %w", err)
+	}
+	return nil
+}
```

Comment:
I think it might be possible to simplify some of the test setup/teardown by using the CLI as a library. Right now this doesn't make me want to write more of these style tests 😓 

Eg.

```go
ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
defer cancel()

go func() {
	if err := buildCLI().RunContext(ctx, []string{"start", "--ephemeral", "--config", "."}); err != nil {
		t.Fatal(err)
	}
}()

// Run some tests
// ...
```

If we dropped that test into the tls example directory, I _think_ it would work without any `runtime` shenanigans.

Reply:
:+1: WIll try

Reply:
> If we dropped that test into the tls example directory

Getting:

> import "github.com/temporalio/temporalite/cmd/temporalite" is a program, not an importable package

Just gonna leave test in cmd dir for now

Reply:
Yeah I'd be up for moving the CLI to an internal package eventually. I guess we should be able to use a relative path to get to the tls example config in the mean time.

Reply:
Unfortunately I will be making a dynamic YAML because it is unreasonable for me to set the working directory of tests and use relative paths to the certs inside my config. Oh well.

Reply:
Done using temp dir and templated config. Probably fine for now, will want a more robust harness w/ helpers if we make more of these.